homeassistant/components/lcn/device_trigger.py

```diff
@@ -0,0 +1,106 @@
+from homeassistant.helpers.typing import ConfigType
+
+from . import DOMAIN
+from .const import KEY_ACTIONS, SENDKEYS
+
+TRIGGER_TYPES = {"transmitter", "transponder", "fingerprint", "send_keys"}
+
+LCN_DEVICE_TRIGGER_BASE_SCHEMA = DEVICE_TRIGGER_BASE_SCHEMA.extend(
+    {vol.Required(CONF_TYPE): vol.In(TRIGGER_TYPES)}
+)
+
+ACCESS_CONTROL_SCHEMA = {vol.Optional("code"): vol.All(vol.Lower, cv.string)}
+TRANSMITTER_SCHEMA = {
+    **ACCESS_CONTROL_SCHEMA,
+    vol.Optional("level"): cv.positive_int,
+    vol.Optional("key"): cv.positive_int,
+    vol.Optional("action"): vol.In([action.lower() for action in KEY_ACTIONS]),
+}
+
+SENDKEYS_SCHEMA = {
+    vol.Optional("key"): vol.In([key.lower() for key in SENDKEYS]),
+    vol.Optional("action"): vol.In([action.lower() for action in KEY_ACTIONS]),
+}
+
+TRIGGER_SCHEMA = vol.Any(
+    LCN_DEVICE_TRIGGER_BASE_SCHEMA.extend(ACCESS_CONTROL_SCHEMA),
+    LCN_DEVICE_TRIGGER_BASE_SCHEMA.extend(TRANSMITTER_SCHEMA),
+    LCN_DEVICE_TRIGGER_BASE_SCHEMA.extend(SENDKEYS_SCHEMA),
+)
+
+
+async def async_get_triggers(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, Any]]:
+    """List device triggers for LCN devices."""
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get(device_id)
+
+    if device.model.startswith(("LCN host", "LCN group", "LCN resource")):  # type: ignore[union-attr]
+        return []
+
+    base_trigger = {
+        CONF_PLATFORM: "device",
+        CONF_DOMAIN: DOMAIN,
+        CONF_DEVICE_ID: device_id,
+    }
+
+    return [{**base_trigger, CONF_TYPE: type_} for type_ in TRIGGER_TYPES]
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: AutomationActionType,
+    automation_info: AutomationTriggerInfo,
+) -> CALLBACK_TYPE:
+    """Attach a trigger."""
+    event_data = {CONF_DEVICE_ID: config[CONF_DEVICE_ID]}
+    event_data.update(
+        {
+            key: config[key]
+            for key in ("code", "level", "key", "action")
+            if key in config
+        }
+    )
+
+    event_config = event.TRIGGER_SCHEMA(
+        {
+            event.CONF_PLATFORM: "event",
+            event.CONF_EVENT_TYPE: f"lcn_{config[CONF_TYPE]}",
+            event.CONF_EVENT_DATA: event_data,
+        }
+    )
+
+    return await event.async_attach_trigger(
+        hass, event_config, action, automation_info, platform_type="device"
+    )
+
+
+async def async_get_trigger_capabilities(hass: HomeAssistant, config: dict) -> dict:
+    """List trigger capabilities."""
+    if config[CONF_TYPE] == "transmitter":
+        return {"extra_fields": vol.Schema(TRANSMITTER_SCHEMA)}
+    if config[CONF_TYPE] == "transponder":
+        return {"extra_fields": vol.Schema(ACCESS_CONTROL_SCHEMA)}
+    if config[CONF_TYPE] == "fingerprint":
+        return {"extra_fields": vol.Schema(ACCESS_CONTROL_SCHEMA)}
+    if config[CONF_TYPE] == "send_keys":
+        return {"extra_fields": vol.Schema(SENDKEYS_SCHEMA)}
+    return {}
```

Comment:
Avoid the linear search by making this a dict
```
TYPE_SCHEMA = {
   "transmitter": {"extra_fields": vol.Schema(TRANSMITTER_SCHEMA)},
   "transponder": .....
}
```

Then you can do
``
return TYPE_SCHEMA.get(config[CONF_TYPE], {})
```